### ISSUE TITLE
Snes mini "CANOE" and Nes mini "KACHIKACHI" command Line arguments testing

### ISSUE THREAD

**Issue by codex785** [ClusterM/hakchi2#478]
Hi, 
is it possible to create a section in hakchi/github with working command lines for both emulators,
so we can test parameters and share experience.

My personnal opinion is that these command lines do exist for some reasons and NERD team has created a "link-the-dots" challenge for dev/hackers in order to have fully working emulators.

**Comment by gingerbeardman** [comment]
Are the binaries encrypted? I doubt it. 

So you should be able to use `strings` on the command line on a computer to figure out what parameters the apps might support.

**Comment by ClusterM** [comment]
I'm using with canoe:

    -boost-fx FACTOR                      Multiply SuperFX clock speed
    -filter INTEGER                       Activate a post-process graphics filter (0=None 1=OpenGL 2=Scanlines 3=CRT)
    -magfilter INTEGER                    Specifies the magnification filter (0=Nearest 1=Linear 2=HorizontalLinear 3=AntiAliasedNearest). The default is nearest with -filter 1 and linear with -filter 2 or -filter 3.

**Comment by codex785** [comment]
Here is what i found interresting:

1673992:39RomPatches_NotYetSplitByGameAndCategory
1674040:19RomPatches_StarFox2
1674064:18RomPatches_StarFox
1674088:23RomPatches_StarFox_1or2
1674128:21RomPatches_FireEmblem
1674152:25RomPatches_KirbySuperStar
1674184:N4SHVC12SA1_CPU65816INS_7SA1CoreEEE
1674232:N4SHVC7SA1CoreE
1674256:N4nerd5ArmetILi5ELi5EEE
1674280:N4nerd11concurrency8WaitableE
1674320:N4nerd6replay5AgentE
1674353:PNG
1674412: "(*
1674436:Dv(
1685076: v(
1694184:    !!!!""""####$$$$%%%%&&&&''''(((())))****++++,,,,----....////0000111122223333444455556666777788889999::::;;;;<<<<====>>>>????
1694344:RendererSdlDisplay
1694368:--version
1694380:Show version then exit
1694404:TERM
1694412:Fiber support needs the thread-local-storage to be available.
1694476:TimerQueue
1694488:-output-dir
1694500:DIRECTORY
1694512:Specify where output files are written
1694552:-help
1694560:Display this help screen
1694588:-re
1694596:Resume state and play all .inputs in turn.
1694640:-resume
1694648:FILENAME.break
1694664:Load an emulator state snapshot from specified file
1694716:-replay
1694724:FILENAME.inputs
1694740:Process recorded user inputs in the specified file
1694792:-replay-all
1694804:Continue processing all .input files in sequential order
1694864:-rollback-mode
1694880:MODE
1694888:Start in rollback mode (0=Idle 1=Record 2=Replay)
1694940:-rollback-mode: invalid value %u
1694976:-rollback-snapshot-period
1695004:FRAMES
1695012:Save a rollback snapshot every N frames (60 for 1 second)
1695072:-rollback-input-dir
1695092:Specify where rollback files are read from
1695136:-rollback-output-dir
1695160:Specify where rollback files are written
1695204:-rollback-discard-data
1695228:Discard input rollback data when starting recording instead of appending to it
1695308:-during
1695316:SECONDS
1695324:Exit after given guest seconds elapsed
1695364:-during-frames
1695380:Exit after given guest ticks
1695412:--save-screenshot-on-quit
1695440:PATH
1695448:Save host screenshot before exiting
1695484:--volume
1695496:PERCENT
1695504:Set the master volume
1695528:-no-audio
1695540:Do not open nor output audio
1695572:-boost-fx
1695584:FACTOR
1695592:Multiply SuperFX clock speed
1695624:Number expected after %s
1695652:-no-lowlatency
1695668:Render in a separate thread, to accommodate "slow" titles.
1695728:-lowlatency
1695740:Render on the main thread to reduce input latency.
1695792:-no-cpurender
1695808:Use the old GPU code for rendering
1695844:-cpurender
1695856:Use the CPU for rendering
1695884:-filter
1695892:INTEGER
1695900:Activate a post-process graphics filter (0=None 1=OpenGL 2=Scanlines 3=CRT)
1695976:-magfilter
1695988:Specifies the magnification filter (0=Nearest 1=Linear 2=HorizontalLinear 3=AntiAliasedNearest). The default is nearest with -filter 1 and linear with -filter 2 or -filter 3.
1696164:--wait-transition-fd
1696188:EVENTFD
1696196:Specifies the event fd to read before drawing the first frame.
1696260:--start-transition-fd
1696284:Specifies the event fd write to when starting the exit to menu transition.
1696360:--finish-transition-fd
1696384:Specifies the event fd to write when the exit to menu transition is finished.
1696464:--transition-to-menu-sprite
1696492:FILENAME
1696504:Specifies the menu transition sprite sheet to use.
1696556:--transition-to-menu-at-exit
1696588:Plays the transition animation when exiting.
1696636:--transition-from-menu
1696660:Plays the transition animation before starting.
1696708:-fp
1696712:Activate Flash/Patterns compensation filter (0=None, 1=VcPhoto 2-5=Armet:Additive,Blend,MonoAdditive 100=auto)
1696824:-glFinish
1696836:Graphics option to reduces latency on mali400, but may degrade framerate
1696912:-no-glFinish
1696928:Opposite of the above option, which became default as of 1.9.1201
1696996:-render-soft
1697012:Use SDL software renderer.
1697040:-show-minimized
1697056:Create window in background
1697084:--use-decorative-frame
1697108:Show a decorative frame using the given file, minus the extension
1697176:--decorative-frame-hue
1697200:Automatically tint the decorative frame, using the game's output
1697268:--decorative-frame-saturation
1697300:Automatically desaturate decorative frame, using the game's output
1697368:--decorative-frame-luminosity
1697400:Automatically change the decorative frame's luminosity, using the game's output
1697480:--pixel-perfect
1697496:Render the screen using square pixels
1697536:--rollback-ui
1697552:Load the rollback UI from this folder
1697592:-record-next
1697608:Save next state snapshot and inputs on exit
1697656:%02d
1697664:--save-on-quit
1697680:Save state snapshot on exit in given filename
1697728:-use-state-saver
1697748:Use StateSaver instead of QuickSave
1697784:--sram-file
1697796:FILENAME.sfrom
1697812:Path to save the SRAM to when it's modified
1697856:-exit-on-sram-file-load-error
1697888:Exit if loading the SRAM file (--sram-file) failed
1697940:--enable-sram-file-hash
1697964:Output a .hash file for SRAM files (standalone and save state)
1698028:--enable-sram-file-hash: cannot be used without --sram-file, --save-on-quit or -rollback-mode
1698124:--load-time-path
1698144:Path to load the playtime
1698176:Loading %s %s
1698192:Play Time file
1698208:Couldn't load %s %s
1698232:--save-time-path
1698252:Path to save the playtime
1698280:-rom
1698288:The game to emulate
1698312:Missing argument %s
1698336:Unknown argument %d: %s
1698364:Usage: %s [options]
1698384:where options are:
1698404:%s %s%s%s
1698416:
1698460:Couldn't load ROM %s
1698484:Bad .sfrom format !
1698508:%02x
1698516:# RomMd5:
1698528:Canoe-SHVC 2.2.0426.0
1698552:%03d
1698560:Missing value after %s
1698584:%%0%dd
1698592:Integer expected after %s
1698620:%15s
1698628:%s.png
1698640:xterm
1698648:xterm-color
1698660:xterm-256color
1698676:screen
1698684:screen-256color
1698700:linux
1698929:        ((((((((0000000088888888@@@@@@@@HHHHHHHHPPPPPPPPXXXXXXXX````````hhhhhhhhppppppppxxxxxxxx
1701696:11ReplayInput
1701744:N4nerd11concurrency5EventE
1706048:    @@@@````
1706576:@ADEPQTU
1707368:N4SHVC3SA1E
1707428:   ""
1707436:" ""
1707520:10RomPatches
1709652:@@@DD
1709660:D@DDN4nerd11concurrency6ThreadE
1714491:H|e?v
1714501: =5_I
1714513:2?G
1714525:58B
1714553:8K5
1714569:!N!

**Comment by gingerbeardman** [comment]
Seems that you can run that app with -help for it to display available options?

**Comment by codex785** [comment]
**# _Here it is:_**

Usage: canoe-shvc [options]
where options are:
--version                             Show version then exit
-output-dir DIRECTORY                 Specify where output files are written
-help                                 Display this help screen
-re 07                                Resume state and play all .inputs in turn.
-resume FILENAME.break                Load an emulator state snapshot from specified file
-replay FILENAME.inputs               Process recorded user inputs in the specified file
-replay-all                           Continue processing all .input files in sequential order
-rollback-mode MODE                   Start in rollback mode (0=Idle 1=Record 2=Replay)
-rollback-snapshot-period FRAMES      Save a rollback snapshot every N frames (60 for 1 second)
-rollback-input-dir DIRECTORY         Specify where rollback files are read from
-rollback-output-dir DIRECTORY        Specify where rollback files are written
-rollback-discard-data                Discard input rollback data when starting recording instead of appending to it
-during SECONDS                       Exit after given guest seconds elapsed
-during-frames FRAMES                 Exit after given guest ticks
--save-screenshot-on-quit PATH        Save host screenshot before exiting
--volume PERCENT                      Set the master volume
-no-audio                             Do not open nor output audio
-boost-fx FACTOR                      Multiply SuperFX clock speed
-no-lowlatency                        Render in a separate thread, to accommodate "slow" titles.
-lowlatency                           Render on the main thread to reduce input latency.
-no-cpurender                         Use the old GPU code for rendering
-cpurender                            Use the CPU for rendering
-filter INTEGER                       Activate a post-process graphics filter (0=None 1=OpenGL 2=Scanlines 3=CRT)
-magfilter INTEGER                    Specifies the magnification filter (0=Nearest 1=Linear 2=HorizontalLinear 3=AntiAl
iasedNearest). The default is nearest with -filter 1 and linear with -filter 2 or -filter 3.
--wait-transition-fd EVENTFD          Specifies the event fd to read before drawing the first frame.
--start-transition-fd EVENTFD         Specifies the event fd write to when starting the exit to menu transition.
--finish-transition-fd EVENTFD        Specifies the event fd to write when the exit to menu transition is finished.
--transition-to-menu-sprite FILENAME  Specifies the menu transition sprite sheet to use.
--transition-to-menu-at-exit          Plays the transition animation when exiting.
--transition-from-menu                Plays the transition animation before starting.
-fp INTEGER                           Activate Flash/Patterns compensation filter (0=None, 1=VcPhoto 2-5=Armet:Additive,
Blend,MonoAdditive 100=auto)
-glFinish                             Graphics option to reduces latency on mali400, but may degrade framerate
-no-glFinish                          Opposite of the above option, which became default as of 1.9.1201
-render-soft                          Use SDL software renderer.
-show-minimized                       Create window in background
--use-decorative-frame PATH           Show a decorative frame using the given file, minus the extension
--decorative-frame-hue                Automatically tint the decorative frame, using the game's output
--decorative-frame-saturation         Automatically desaturate decorative frame, using the game's output
--decorative-frame-luminosity         Automatically change the decorative frame's luminosity, using the game's output
--pixel-perfect                       Render the screen using square pixels
--rollback-ui PATH                    Load the rollback UI from this folder
-record-next                          Save next state snapshot and inputs on exit
--save-on-quit FILENAME               Save state snapshot on exit in given filename
-use-state-saver                      Use StateSaver instead of QuickSave
--sram-file FILENAME.sfrom            Path to save the SRAM to when it's modified
-exit-on-sram-file-load-error         Exit if loading the SRAM file (--sram-file) failed
--enable-sram-file-hash               Output a .hash file for SRAM files (standalone and save state)
--load-time-path FILENAME.sfrom       Path to load the playtime
--save-time-path FILENAME.sfrom       Path to save the playtime
-rom FILENAME.sfrom                   The game to emulate

**Comment by barryvdh** [comment]
Did anyone try the flags to see if they improve compatibility with certain games, like the GPU renderer or lowlatency options?

**Comment by gingerbeardman** [comment]
See here: http://www.neogaf.com/forum/showthread.php?p=251465186&highlight=lowlatency#post251465186

**Comment by Drakonas** [comment]
This has been added to the wiki as of writing. Closing.
